### PR TITLE
[NO SQUASH] Cleanup, add luacheck, fix nil error

### LINKED ISSUE #6: Torch rockets are causing game to kill itself
Every time I try to use a torch rocket I get this error:

2023-10-22 17:59:53: ERROR[Main]: ServerError: AsyncErr: Lua: Runtime error from mod 'torch_bomb' in callback luaentity_Step(): D:\Games\Minetest\bin\..\mods\torch_bomb\init.lua:286: attempt to index local 'above_def' (a nil value)
2023-10-22 17:59:53: ERROR[Main]: stack traceback:
2023-10-22 17:59:53: ERROR[Main]: 	D:\Games\Minetest\bin\..\mods\torch_bomb\init.lua:286: in function 'find_target'
2023-10-22 17:59:53: ERROR[Main]: 	D:\Games\Minetest\bin\..\mods\torch_bomb\init.lua:336: in function 'kerblam'
2023-10-22 17:59:53: ERROR[Main]: 	D:\Games\Minetest\bin\..\mods\torch_bomb\init.lua:498: in function 'entity_detonate'
2023-10-22 17:59:53: ERROR[Main]: 	D:\Games\Minetest\bin\..\mods\torch_bomb\init.lua:556: in function <D:\Games\Minetest\bin\..\mods\torch_bomb\init.lua:533>

### PR DESCRIPTION
**Pull request by Niklp09**
- Cleanup/remove old [translation] files
- Add luacheck and fix warnings (uses mt-mods [zero-maintenance workflow](https://github.com/mt-mods/mt-mods/blob/master/snippets/luacheck.yml))
- Bump minimal minetest version to 5.3 (the mod uses now `minetest.is_creative_enabled()`
- Fix https://github.com/minetest-mods/torch_bomb/issues/6

### CODE REVIEW

File: mod.conf [hunk 1]
```diff
@@ -1,4 +1,4 @@
 name = torch_bomb
 description = Place torches throughout your entire surroundings with a torch bomb
-depends =
-optional_depends = mcl_core, mcl_sounds, mcl_explosions, mcl_tnt, default, tnt, creative, farming
\ No newline at end of file
+optional_depends = mcl_core, mcl_sounds, mcl_explosions, mcl_tnt, default, tnt, creative, farming
+min_minetest_version = 5.3.0
```
Comment (SmallJoker) [inline mod.conf:4]: Why 5.3.0?
EDIT: As in: Is there a particular reason or just a wild guess?

Comment (Niklp09) [reply mod.conf:4]: as mentioned in the PR description:
> Bump minimal minetest version to 5.3 (the mod uses now minetest.is_creative_enabled())

mt 5.3.0 is the first version w/ this function.

Comment (SmallJoker) [reply mod.conf:4]: I see. Thank you for pointing this out, I seem to have skipped right to the diff tab.

**Comment by SmallJoker** [comment]
:+1: Gave it a quick test. Works. The cleanup also looks pretty trivial.

Will merge in a few days unless there are objections.

**Pull request merged**